### ISSUE TITLE
RM links against C++ API of OSVR Core, requiring runtime, compiler match

### ISSUE THREAD

**Issue by rpavlik** [sensics/OSVR-RenderManager#193]
The logging code is currently linking against the OSVR-Core osvrUtil library's internal C++ API to get the "full" logging API. Unfortunately, this means that RM and Core must be built by the same compiler, in the same configuration, against the same runtime library.

I've stubbed out an hourglass interface to solve this problem (putting the C++ API behind a C one, then wrapping the C API with a C++ header-only API that strongly resembles the original API) here: https://github.com/OSVR/OSVR-Core/tree/c-api-for-full-logging - it just needs implementation.

cc @JeroMiya

**Comment by russell-taylor** [comment]
Or we could pull RM into the OSVR-Core structure, putting it under osvr/renderkit, and then compile it all at once...

**Comment by rpavlik** [comment]
yes, but a C API would make it less fragile anyway, and is faster to do.
